src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -41,39 +43,44 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             PatternSyntax node,
             TypeSymbol operandType,
             bool hasErrors,
-            DiagnosticBag diagnostics,
-            bool wasSwitchCase = false)
+            DiagnosticBag diagnostics)
         {
             switch (node.Kind())
             {
+                case SyntaxKind.DiscardPattern:
+                    return BindDiscardPattern((DiscardPatternSyntax)node, operandType, hasErrors, diagnostics);
+
                 case SyntaxKind.DeclarationPattern:
-                    return BindDeclarationPattern(
-                        (DeclarationPatternSyntax)node, operandType, hasErrors, diagnostics);
+                    return BindDeclarationPattern((DeclarationPatternSyntax)node, operandType, hasErrors, diagnostics);
 
                 case SyntaxKind.ConstantPattern:
-                    return BindConstantPattern(
-                        (ConstantPatternSyntax)node, operandType, hasErrors, diagnostics, wasSwitchCase);
+                    return BindConstantPattern((ConstantPatternSyntax)node, operandType, hasErrors, diagnostics);
 
                 case SyntaxKind.DeconstructionPattern:
+                    return BindDeconstructionPattern((DeconstructionPatternSyntax)node, operandType, hasErrors, diagnostics);
+
                 case SyntaxKind.PropertyPattern:
-                    // PROTOTYPE(patterns2): binding not yet supported for recursive pattern forms.
-                    diagnostics.Add(ErrorCode.ERR_BindToBogus, node.Location, "recursive pattern");
-                    return new BoundConstantPattern(node, BadExpression(node), ConstantValue.Null, hasErrors: true);
+                    return BindPropertyPattern((PropertyPatternSyntax)node, operandType, hasErrors, diagnostics);
 
                 default:
                     throw ExceptionUtilities.UnexpectedValue(node.Kind());
             }
         }
 
+        private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol operandType, bool hasErrors, DiagnosticBag diagnostics)
+        {
+            // TODO(patterns2): give an error if there is a bindable `_` in scope.
```

Comment:
Shouldn't this say "PROTOTYPE"?

Reply:
Recorded in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222); will be addressed in a future PR. I will point to the specific resolving delta in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222).

---
In reply to: [154579242](https://github.com/dotnet/roslyn/pull/23209#discussion_r154579242) [](ancestors = 154579242)